src/Orleans/Utils/IKeyedServiceCollection.cs

```diff
@@ -0,0 +1,37 @@
+﻿
+using System;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Orleans
```

Comment:
Even if not completely right either, I would prefer we put this in Orleans.Runtime as other utilities. Putting it in the root namespace would mean that this extension method should be discoverable by all usages of Orleans, where in fact this is not Orleans' specific, and we probably don't expect users to care about it, only extension implementations would.